File: src/go/svd/models/cargo_information.go
package models

import "time"

// CargoInformation represents the "Cargo" section of SVD.
type CargoInformation struct {
	CargoDescription             string    `json:"cargoDescription" xml:"CargoDescription" csv:"CargoDescription"`
	GrossWeight                  float64   `json:"grossWeight" xml:"GrossWeight" csv:"GrossWeight"`
	GrossVolume                  float64   `json:"grossVolume" xml:"GrossVolume" csv:"GrossVolume"`
	BillOfLadingReference        string    `json:"billOfLadingReference" xml:"BillOfLadingReference" csv:"BillOfLadingReference"`
	BillOfLadingIssuedDate       time.Time `json:"billOfLadingIssuedDate" xml:"BillOfLadingIssuedDate" csv:"BillOfLadingIssuedDate"`
	TotalContainersTEU           int       `json:"totalContainersTEU" xml:"TotalContainersTEU" csv:"TotalContainersTEU"`
	TotalFullContainersTEU       int       `json:"totalFullContainersTEU" xml:"TotalFullContainersTEU" csv:"TotalFullContainersTEU"`
	TotalFullReeferContainersTEU int       `json:"totalFullReeferContainersTEU" xml:"TotalFullReeferContainersTEU" csv:"TotalFullReeferContainersTEU"`
	ReeferSocketsInUse           int       `json:"reeferSocketsInUse" xml:"ReeferSocketsInUse" csv:"ReeferSocketsInUse"`
	Chilled20FtReeferContainers  int       `json:"chilled20FtReeferContainers" xml:"Chilled20FtReeferContainers" csv:"Chilled20FtReeferContainers"`
	Chilled40FtReeferContainers  int       `json:"chilled40FtReeferContainers" xml:"Chilled40FtReeferContainers" csv:"Chilled40FtReeferContainers"`
	Frozen20FtReeferContainers   int       `json:"frozen20FtReeferContainers" xml:"Frozen20FtReeferContainers" csv:"Frozen20FtReeferContainers"`
	Frozen40FtReeferContainers   int       `json:"frozen40FtReeferContainers" xml:"Frozen40FtReeferContainers" csv:"Frozen40FtReeferContainers"`
	TotalVehiclesCEU             int       `json:"totalVehiclesCEU" xml:"TotalVehiclesCEU" csv:"TotalVehiclesCEU"`
}

File: src/elixir/svd/lib/svd/models/cargo_information.ex
defmodule SVD.Models.CargoInformation do
  @moduledoc false

  @fields [
    :cargo_type,
    :cargo_description,
    :cargo_quantity,
    :cargo_loaded,
    :cargo_discharged,
    :cargo_rob,
    :teu_count,
    :feu_count,
    :passenger_count,
    :vehicle_count,
    :gross_weight,
    :gross_volume,
    :bill_of_lading_reference,
    :bill_of_lading_issued_date,
    :total_containers_teu,
    :total_full_containers_teu,
    :total_full_reefer_containers_teu,
    :reefer_sockets_in_use,
    :chilled_20_ft_reefer_containers,
    :chilled_40_ft_reefer_containers,
    :frozen_20_ft_reefer_containers,
    :frozen_40_ft_reefer_containers,
    :total_vehicles_ceu
  ]

  @derive {Jason.Encoder, only: @fields}
  defstruct @fields

  @type t :: %__MODULE__{}

  @spec fields() :: [atom()]
  def fields, do: @fields
end
